File: Neetcode/TwoSum2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l=0
        r=len(numbers)-1
        sm = numbers[l] + numbers[r]
        while sm != target:
            if sm < target:
                l += 1
                while numbers[l] == numbers[l-1]:
                    l += 1
            elif sm > target:
                r -= 1
                while numbers[r] == numbers[r+1]:
                    r -= 1
            sm = numbers[l] + numbers[r]
        return [l+1, r+1]

File: Neetcode/TwoSum2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0;
        int j = numbers.size() - 1;
        int s = 0;
        while (i<j){
            s = numbers[i] + numbers[j];
            if (s == target){
                return {i+1, j+1};
            }
            else if (s < target){
                i += 1;
            }
            else{
                j -= 1;
            }
        }
        return {-1, -1};
    }
};
